### ISSUE TITLE
NaiveBayesClassifier taking too long

### ISSUE THREAD

**Issue by canivel** [sloria/TextBlob#63]
Hi, I've a small dataset of 1000 tweets which I've classify in pos/neg for training. When I tried to use it at the NaiveBayesClassifier() it tooks like 10-15min to return a result...
Is there a way to save the result of the classifier like a dump and reuse that for further classifications ?

Thanks

**Comment by shackra** [comment]
I'm experiencing this, but I cannot get rid of the training set because is in French, which means every string is UTF-8 encoded.

I actually created a training set encoded as ASCII, ignoring the characters outside that encoding (which means losing a lot of data), and the training phase was still taking too much time.

What can I do?

**Comment by methodds** [comment]
Hi, I also noticed the TextBlob classifier to take very long in comparison to sklearn and the NLTK-classifier. But I don't understand why there is a difference, especially for NLTK. Isn't TextBlob using the NLTK version?

**Comment by chandra589** [comment]
Can I know how to get the accuracy and most informative features of the naive bayes classifier which is already trained using movie corpus.

Thanks

**Comment by DSA101** [comment]
Experiencing similar performance issues. Using custom NaiveBayes classifier to train on 1500 article titles (pos, neg, NA). Takes about 10 minutes to train on core i7 and not much faster to classify the same number of titles with a pickled classifier (loading is fast, but classification is slow). Any ideas how to speed this up? I can understand that training is slow, but why is classification slow too?

**Comment by jcalbert** [comment]
I ran into the same trouble.  It seems that `basic_extractor` re-tokenizes the training set with every update AND for each item in the original training set (so training is O(n^2)).

I'm working on a fix that should also fix #77 and #123.

**Comment by gcmicah** [comment]
any update on this issue?

**Comment by sloria** [comment]
#136 is now merged and released to PyPI.
